gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+      unsigned               pass_number,
+      const semantics_map_t* semantics_map,
+      pass_semantics_t*      out)
+{
+   unordered_map<string, slang_texture_semantic_map> texture_semantic_map;
+   unordered_map<string, slang_texture_semantic_map> texture_semantic_uniform_map;
+
+   string name = shader_info->pass[pass_number].alias;
+
+   if (!set_unique_map(
+             texture_semantic_map, name,
+             slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_PASS_OUTPUT, pass_number }))
+      return false;
+
+   if (!set_unique_map(
+             texture_semantic_uniform_map, name + "Size",
+             slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_PASS_OUTPUT, pass_number }))
+      return false;
+
+   if (!set_unique_map(
+             texture_semantic_map, name + "Feedback",
+             slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_PASS_FEEDBACK, pass_number }))
+      return false;
+
+   if (!set_unique_map(
+             texture_semantic_uniform_map, name + "FeedbackSize",
+             slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_PASS_FEEDBACK, pass_number }))
+      return false;
+
+   for (unsigned i = 0; i < shader_info->luts; i++)
+   {
+      if (!set_unique_map(
+                texture_semantic_map, shader_info->lut[i].id,
+                slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_USER, i }))
+         return false;
+
+      if (!set_unique_map(
+                texture_semantic_uniform_map, string(shader_info->lut[i].id) + "Size",
+                slang_texture_semantic_map{ SLANG_TEXTURE_SEMANTIC_USER, i }))
+         return false;
+   }
+
+   unordered_map<string, slang_semantic_map> uniform_semantic_map;
+   for (unsigned i = 0; i < shader_info->num_parameters; i++)
+   {
+      if (!set_unique_map(
+                uniform_semantic_map, shader_info->parameters[i].id,
+                { SLANG_SEMANTIC_FLOAT_PARAMETER, i }))
+         return false;
+   }
+
+   slang_reflection sl_reflection;
+   sl_reflection.pass_number                  = pass_number;
+   sl_reflection.texture_semantic_map         = &texture_semantic_map;
+   sl_reflection.texture_semantic_uniform_map = &texture_semantic_uniform_map;
+   sl_reflection.semantic_map                 = &uniform_semantic_map;
+
+   if (!slang_reflect(*vs_compiler, *ps_compiler, vs_resources, ps_resources, &sl_reflection))
+   {
+      RARCH_ERR("[slang]: Failed to reflect SPIR-V. Resource usage is inconsistent with "
+                "expectations.\n");
+      return false;
+   }
+
+   memset(out, 0x00, sizeof(*out));
+
+   out->cbuffers[SLANG_CBUFFER_UBO].stage_mask = sl_reflection.ubo_stage_mask;
+   out->cbuffers[SLANG_CBUFFER_UBO].binding    = sl_reflection.ubo_binding;
+   out->cbuffers[SLANG_CBUFFER_UBO].size       = (sl_reflection.ubo_size + 0xF) & ~0xF;
+   out->cbuffers[SLANG_CBUFFER_PC].stage_mask  = sl_reflection.push_constant_stage_mask;
+   out->cbuffers[SLANG_CBUFFER_PC].binding     = sl_reflection.ubo_binding ? 0 : 1;
```

Comment:
I don't understand this. For HLSL you'll have to invent a cbuffer binding for the push constant block, but this can probably alias with the UBO block?

Reply:
the push_constant cbuffer doesn't have an explicit binding, so it will just take the first available slot. which is either 0 or 1 depending on the UBO binding.

Reply:
Do you remap the UBO binding to 1 in that case before calling CompilerHLSL::compile then?

Reply:
no I don't need to do an explicit remapping, I just guess the push_constant binding.
if: UBO is bound to 0 --> push_constant is bound to 1, else: push_constant is bound to 0.